packages/gestalt/src/ModuleTitle.js

```diff
@@ -1,62 +1,64 @@
 // @flow strict
-import { type Node } from 'react';
+import type { Node } from 'react';
 import Badge from './Badge.js';
 import Box from './Box.js';
 import Flex from './Flex.js';
 import Icon from './Icon.js';
 import Text from './Text.js';
-import { type BaseModuleTitleProps, type TypeOptions } from './moduleTypes.js';
-
-type Props = {|
-  ...BaseModuleTitleProps,
-  title: string, // overwriting to be required
-  type: TypeOptions, // overwriting to be required
-|};
+import type { ModuleTitleProps } from './moduleTypes.js';
 
 /**
  * https://gestalt.pinterest.systems/Module
  */
-export default function ModuleTitle({
-  badgeText,
-  icon,
-  iconAccessibilityLabel,
-  title,
-  type = 'info',
-}: Props): Node {
+export default function ModuleTitle(props: ModuleTitleProps): Node {
+  const { title, type = 'info' } = props;
+
   const TYPE_ICON_ATTRIBUTES = {
-    info: {
-      icon,
-      color: 'darkGray',
-    },
+    info: props.icon
+      ? {
+          icon: props.icon,
+          color: 'darkGray',
+          accessibilityLabel: props.iconAccessibilityLabel,
+        }
+      : {},
     error: {
       icon: 'workflow-status-problem',
       color: 'red',
+      accessibilityLabel: props.iconAccessibilityLabel,
     },
   };
 
-  const { color, icon: iconName } = TYPE_ICON_ATTRIBUTES[type];
+  const { accessibilityLabel = '', color, icon: iconName } = TYPE_ICON_ATTRIBUTES[type];
 
   return (
-    <Flex gap={2}>
+    <Flex alignItems="center" gap={2}>
       {iconName && (
-        <Icon accessibilityLabel={iconAccessibilityLabel ?? ''} color={color} icon={iconName} />
+        <Flex.Item minWidth={0}>
+          <Icon accessibilityLabel={accessibilityLabel} color={color} icon={iconName} />
+        </Flex.Item>
       )}
 
-      <Flex.Item minWidth={0}>
-        <Text color={color} lineClamp={1} weight="bold">
-          {title}
-        </Text>
-      </Flex.Item>
+      {title && (
```

Comment:
I think we'll need stricter enforcement than this, as the user can add a `$FlowFixMe` and add all three possibilities. We also want to make sure that even if the user passes, say, both iconButton and badgeText, we only actually show one of them.

Reply:
Thanks @dangerismycat , I just added something to prevent they all rendering together